integration-tests/api/__tests__/store/cart.js

```diff
@@ -47,8 +49,13 @@ describe("/store/carts", () => {
   beforeAll(async () => {
     const cwd = path.resolve(path.join(__dirname, "..", ".."))
     try {
-      dbConnection = await initDb({ cwd })
-      medusaProcess = await setupServer({ cwd, verbose: false })
+      const [process, connection] = await startServerWithEnvironment({
```

Comment:
todo: The existing integration tests should not be impacted by the sales channels flag. Could you add some integrations tests and place them in the right sales channels tests suite. You can look at my pr to see how i did it